### ISSUE TITLE
Retention-by-exclusion

### ISSUE THREAD

**Issue by nblumhardt** [datalust/seq-tickets#860]
In some cases it has been reported that modelling retention policies as "everything except X" would simplify configuration.

It's possible that we could do this by keeping two kinds of retention policy: those that specify signals to delete, as we do today, and those that specify signals to retain.

For example, "After 5 days, delete everything except: Errors, Audit, Transactions".

Raising this ticket for further discussion/consideration.

**Comment by nblumhardt** [comment]
We're still interested in supporting this but won't be able to schedule time for it in the next few releases; we'll revisit down the track.

**Comment by AndrewSav** [comment]
Hello @nblumhardt, could you please clarify, why are you closing this issue if you want to "revisit down the track"? That is what is that your goal, that closing this issue achieves? Keeping it open would achieve that we have a visibility of this as an open issue, and if it won't fall down the cracks after "the next few releases".

**Comment by nblumhardt** [comment]
Hi Andrew, thanks for the note.

My comment above is not entirely accurate, I should have also added, we think #853 would implicitly cover this, and it's still open/being actively tracked. If signal expressions support negation, retention-by-exclusion will fall out from that naturally, and is the likely implementation pathway for this feature.

In general, though, we try to keep the GitHub tracker down to what we can reasonably expect to consider for the N+1 and sometimes +2 releases, but this isn't the full extent of our organisational memory. Closed issues like this one give us something to revisit and review when we look at an area; they're not "won't do" but more like "in cold storage", and this helps us avoid "backlog bankruptcy" wherein the backlog grows so big we can't maintain it effectively.

It's good to know you are still interested in this one. Following/moving the discussion to #853 will keep it on the radar.

Thanks again for jumping in; hope this helps!
Nick

**Comment by AndrewSav** [comment]
Hello @nblumhardt, so not when 853 is also closed, is this now implemented?

**Comment by nblumhardt** [comment]
Hi Andrew! Thanks for the note.

Unfortunately, when we first tried it out, "convert to Discussion" just moved the issue (the old issue URL would redirect, and state wouldn't change). It seems since then, the feature was updated and conversion now ends up closing the original and copying its contents into the discussion (urgh).

This is still being tracked, negatable signals is one of my personal favourites :-) - it's at: https://github.com/datalust/seq-tickets/discussions/1104

We're changing the workflow in this repository to keep pretty much everything in discussions until we know what the implementation approach will be, and are ready to assign it to one of the next few releases. It avoids us having to make arbitrary decisions to close things simply to keep open issues to a reviewable/manageable size.

Cheers!
